### ISSUE TITLE
Registering

### ISSUE THREAD

**Issue by ezdavis** [kiddailey/aspnetimage#4]
I'm in need of this very product. I have lots of sites that used aspimage on iis6 and now I need them on iis8. I've signed the dll in Visual Studio and then copied it onto the server. I then run the regasm from the .net framework on the dll and get a message saying types registered successfully for both /tlb and /codebase, as shown in ASPNetImage1.jpg. However, when I try to run my update for my admin page i recieve the error (ASPNetImage2.jpg) that basically says that it's not registered. Line 71 that it returns the error on is "Set objImage = Server.CreateObject("ASPNetImage.NetImage")". Any suggestions on what the problem could be?
![aspnetimage1](https://cloud.githubusercontent.com/assets/7422825/11593150/acf3cf48-9a70-11e5-813a-f2ac34fe79f6.JPG)
![aspnetimage2](https://cloud.githubusercontent.com/assets/7422825/11593149/acf332e0-9a70-11e5-9a79-368549f58c9f.JPG)

**Comment by kiddailey** [comment]
Looking at your first image, there is no /codebase param on the regasm command-line. Make sure you include both.  See the example register.bat in the Resource folder:

```
c:\Windows\Microsoft.NET\Framework\v2.0.50727\regasm.exe ..\bin\release\ASPNetImage.dll /tlb /codebase
```

**Comment by kiddailey** [comment]
Oh ... looking at your first image, there is no /codebase param on the regasm command-line.  Make sure you include both.

**Comment by ezdavis** [comment]
I ran the command with codebase and then tlb. Does this make a difference from running both of them in the same command? I can test this when I get back to work Monday.

**Comment by kiddailey** [comment]
Quite possibly it may not work when ran separately.  I've not tested it myself, but I've always ran the two together at the same time.

Be sure to unregister it first before attempting to re-register it.

**Comment by ezdavis** [comment]
Okay, so I unregistered it and then registered it again in the same line but my application still gives server.createobject failed/ Any other suggestions? Is there anything I need to do after this on the server side?
![aspnetimage3](https://cloud.githubusercontent.com/assets/7422825/11628001/cae2b30e-9cbc-11e5-965f-618f813b7c23.JPG)

**Comment by ezdavis** [comment]
Am I supposed to run regsvr32 on the dll afterwards? If so, when I try I get this error...
![aspnetimage4](https://cloud.githubusercontent.com/assets/7422825/11628594/b59dc8d6-9cc0-11e5-887e-109f900d184f.JPG)

**Comment by kiddailey** [comment]
Are you compiling 32bit and running with a 32bit app pool?

**Comment by ezdavis** [comment]
Yes, I have my setting to accept 32bit app pool and is set for integrated. Have also tried changing my setting for classic and to not accept 32bit with no success. I can't figure out why it's telling me it's registered but when I run my application I'm getting an error that says it can't find it registered.

**Comment by kiddailey** [comment]
Does the ASPNET user (or whatever your process is running under) have execute permissions for the directory that the DLL is in?  Have you tried putting the DLL in the Windows or System32 folders and registering it there instead?  Have you tried running your command shell as administrator?

**Comment by ezdavis** [comment]
Okay, so when I try to move it to windows\system32 i get an error, but I can run it in windows\syswow64.
I then tested system32 with framework64 and it registered.
![aspnetimage5](https://cloud.githubusercontent.com/assets/7422825/11689671/019aaad2-9e61-11e5-9c6a-9a3ec6ada68d.JPG)
Will this still work on my server?

**Comment by kiddailey** [comment]
Did you try running your app to see if the Server.CreateObject still fails?  Having the DLL in System32 should work fine.

**Comment by ezdavis** [comment]
Yeah I tried it but it still fails

On Dec 9, 2015, at 1:46 PM, kiddailey <notifications@github.com<mailto:notifications@github.com>> wrote:

Did you try running your app to see if the Server.CreateObject still fails? Having the DLL in System32 should work fine.

##

**Comment by Tfx77** [comment]
I too am getting this issue, registered with .net v4.0.30319, then unregistered (and can't find any tracers to aspnetimage in the reg) and then also v2.0.50727 .net. Is it correct that the codebase keys in the register are prefixed with file:/// (probably is). 

Ok, I just looked at the registry entries and I get the following data. 

Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\AspImage.Image]
@="ASPNetImage.NetImage"

[HKEY_CLASSES_ROOT\AspImage.Image\CLSID]
@="{D06078A0-41F0-35ED-8E34-9EBAAB0ED63A}"

So, to call the com object you need to use AspImage.Image (which seems odd, guess I could rename it) as the Classid references all the correct files (in my case D06078A0-41F0-35ED-8E34-9EBAAB0ED63A).

Tested a quick upload and manipulation and it seems to work. 

Is there a way to get this working under 64bit?

**Comment by kiddailey** [comment]
Thanks for the info Tfx77.

In regards to 64bit, I haven't had time to play around with it.  Are you able to recompile the library for that target and try it?

**Comment by Tfx77** [comment]
Not really, I don't know how to do that.
